docfix: "+caliper" variant is now "+profiling"

goxberry opened the pull request:
This commit updates the quickstart documentation regarding building
Serac with Caliper support by replacing the `--spec=+caliper` Serac
Spack package variant with the updated `--spec=+profiling` variant.

goxberry committed:
docfix: "+caliper" variant is now "+profiling"

This commit updates the quickstart documentation regarding building
Serac with Caliper support by replacing the `--spec=+caliper` Serac
Spack package variant with the updated `--spec=+profiling` variant.

white238 commented on src/docs/sphinx/quickstart.rst:
```diff
@@ -145,7 +145,7 @@ Some helpful uberenv options include :
 * ``--spec=+debug`` (build the MFEM and Hypre libraries with debug symbols)
 * ``--spec=+glvis`` (build the optional glvis visualization library)
-* ``--spec=+caliper`` (build the `Caliper performance profiling library <https://github.com/LLNL/Caliper>`_)
+* ``--spec=+profiling`` (build the `Caliper performance profiling library <https://github.com/LLNL/Caliper>`_)
```
This also builds Adiak and would be long in the tooth at that point with both links... 
```suggestion
* ``--spec=+profiling`` (build the Adiak and Caliper libraries)
```

goxberry replied:
Done in [8abf0289](https://github.com/LLNL/serac/pull/652/commits/8abf0289293cdb895311d516f2abb902fe13c6fa).

white238 commented:
Thanks for catching this @goxberry

goxberry committed:
docfix: remove caliper link in variant description

This commit removes the link to Caliper in the `+profiling` variant
description within the quickstart Sphinx documentation, and adds Adiak
to that description, as suggested by Chris White (@white238).

adayton1 reviewed (approved):
Thanks!
